src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,342 @@
+    name: string;
+    alias?: string;
+    align?: TableCellProps["align"];
+    columnCalculation?: "sum" | "average" | ((values: any) => any);
+    editable?: boolean;
+    type?: "text" | "number";
+    numberFormat?: FormatOptions;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<RowData[]>,
+    SettableBoxProps,
+    SettableTableProps {
+    cols: Column[];
+    rows: RowData[];
+    footerLabel?: string;
+    onMouseEnter?: (row: RowData) => void;
+    onMouseLeave?: (row: RowData) => void;
+    onRowEdit?: (row: RowData) => RowData;
+}
+
+//Mimic MUI header style for footer
+const FooterCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    backgroundColor: "var(--secondaryBackground)",
+    fontWeight: "bold",
+    color: "var(--secondaryForeground)",
+    fontSize: "inherit",
+};
+
+//Mimic MUI header style for footer
+const TableCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    fontSize: "inherit",
+};
+
+const InputStyleOverrides: InputProps["sx"] = {
+    ".MuiInput-input": {
+        borderRadius: "4px !important",
+        padding: "5px 8px !important",
+        border: "0px !important",
+        backgroundColor: "inherit",
+        width: "100%",
+        height: "auto",
+    },
+};
+
+const IconButtonStyleOverrides: IconButtonProps["sx"] = {
+    // Hack required to override .gcx-forms.defaults
+    borderRadius: "50% !important",
+    // Hack required to match disabled list item content
+    "&.Mui-disabled": {
+        opacity: 0.38,
+    },
+};
+const calculateFooterRow = (cols: Column[], rows: RowData[]): FooterRow => {
+    const totalCols = cols.filter((x) => x.columnCalculation);
+    const total: FooterRow = {};
+    for (const col of totalCols) {
+        if (col.columnCalculation) {
+            const values = rows.map((x) => x[col.name]);
+            if (col.columnCalculation === "sum") {
+                total[col.name] = calculateSum(values);
+            } else if (col.columnCalculation === "average") {
+                total[col.name] = calculateAverage(values);
+            } else {
+                total[col.name] = col.columnCalculation(values);
+            }
+        }
+    }
+    return total;
+};
+
+const calculateSum = (values: any[]): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    return numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+};
+
+const calculateAverage = (values: any[]): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    const sum = numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+    return sum / numericVals.length || 0;
+};
+
+const formatValue = (value: any, column: Column): any => {
+    if (column.type === "number" && typeof value === "number") {
+        if (column.numberFormat) {
+            return format(column.numberFormat as string, value);
```

Comment:
The `as string` is wrong and not necessary.

Reply:
fixed